modules/citrus-java-dsl/src/main/java/com/consol/citrus/dsl/builder/SendMessageBuilder.java

```diff
@@ -432,6 +439,15 @@ public T messageType(MessageType messageType) {
     public T messageType(String messageType) {
         this.messageType = messageType;
         getAction().setMessageType(messageType);
+
+        if (MessageType.BINARY.name().equals(messageType)) {
+            getMessageContentBuilder().add(binaryMessageConstructionInterceptor);
+        }
+
+        if (MessageType.GZIP.name().equals(messageType)) {
```

Comment:
I think we should use `gzipMessageConstructionInterceptor.supportsMessageType()` here to have one source of truth for the decision, whether the interceptor could handle the message type.